### ISSUE TITLE
Data.Table, Fread

### ISSUE THREAD

**Issue by hansdecocq** [Rdatatable/data.table#2112]
titanic_train  <-  fread("titanic_train.csv", data.table = TRUE)
My_Rows <- c(10:200,250:891)
My_Col    <- c(2:4,6:9)
tmp_1   <- titanic_train[My_Rows,My_Col]    # Works fine

titanic_train   <-  fread("titanic_train.csv", data.table = FALSE)
My_Rows <- c(10:200,250:891)
My_Col   <- c(2:4,6:9)
tmp_1   <- titanic_train[My_Rows,My_Col]    # Produce an error

Error in `[.data.table`(titanic_train, My_Rows, My_Col) : 
  j (the 2nd argument inside [...]) is a single symbol but column name 'My_Col' is not found. Perhaps you intended DT[,..My_Col] or DT[,My_Col,with=FALSE]. 
This difference to data.frame is deliberate and explained in FAQ 1.1.

**Comment by MichaelChirico** [comment]
did you read the FAQ? use with=FALSE.

**Comment by mattdowle** [comment]
Yes.  And not just in the FAQ but in the long and helpful error message too. The error message even includes "Perhaps you intended DT[,..My_Col] or DT[,My_Col,with=FALSE]."  And not only that, but the error message includes "This difference to data.frame is deliberate and explained in FAQ 1.1."  It even points to _which_ FAQ.
@hansdecocq How on earth can we do better than that?

**Comment by hansdecocq** [comment]
Thans for your reply. You're reply "sounded" a bit Irritated. Believe me I took all what's possible to find the proper solution by my self. As a late resort I went to this site. Perhaps a is better explanation in the online r help documentation a solution? Now every one and on every place write people write answers. It took me a while top find your place and was happy to drop my question. As a novice R-user it is difficult to you grasp all possibilities of R. Good explanations of the errors message in general are rare to find in this community. As said before thanks again u will read the section you mentioned. Regards Hans
